test/media-groups.test.js

```diff
@@ -0,0 +1,719 @@
+  const activeTrack = MediaGroups.activeTrack[type](type, settings);
+
+  assert.equal(activeTrack(), null, 'returns null when empty track list');
+
+  tracks.track1 = { id: 'track1', mode: 'disabled' };
+  tracks.track2 = { id: 'track2', mode: 'hidden' };
+  tracks.track3 = { id: 'track3', mode: 'disabled' };
+
+  assert.equal(activeTrack(), null, 'returns null when no active tracks');
+
+  tracks.track3.mode = 'showing';
+
+  assert.strictEqual(activeTrack(), tracks.track3, 'returns active track');
+
+  tracks.track1.mode = 'showing';
+
+  // video.js treats the first enabled track in the track list as the active track
+  // so we want the same behavior here
+  assert.strictEqual(activeTrack(), tracks.track1, 'returns first active track');
+});
+
+QUnit.test('activeGroup returns the correct audio group', function(assert) {
+  const type = 'AUDIO';
+  let media = null;
+  const settings = {
+    mediaTypes: MediaGroups.createMediaTypes(),
+    masterPlaylistLoader: {
+      media: () => media
+    }
+  };
+  const groups = settings.mediaTypes[type].groups;
+  const tracks = settings.mediaTypes[type].tracks;
+  const activeTrack = MediaGroups.activeTrack[type](type, settings);
+  const activeGroup = MediaGroups.activeGroup(type, settings);
+
+  assert.equal(activeGroup(), null, 'returns null when no media in masterPlaylistLoader');
+
+  media = { attributes: { } };
+  groups.main = [{ id: 'en' }, { id: 'fr' }];
+
+  assert.strictEqual(activeGroup(), groups.main,
+    'defaults to main audio group when media does not specify audio group');
+
+  groups.audio = [{ id: 'en'}, { id: 'fr' }];
+  media.attributes.AUDIO = 'audio';
+
+  assert.strictEqual(activeGroup(), groups.audio,
+    'returns list of variants in active audio group');
+
+  tracks.en = { id: 'en', enabled: false };
+  tracks.fr = { id: 'fr', enabled: false };
+
+  assert.equal(activeGroup(activeTrack()), null,
+    'returns null when an active track is specified, but there is no active track');
+
+  tracks.fr.enabled = true;
+
+  assert.strictEqual(activeGroup(activeTrack()), groups.audio[1],
+    'returned the active group corresponding to the active track');
+});
+
+QUnit.test('activeGroup returns the correct subtitle group', function(assert) {
+  const type = 'SUBTITLES';
+  let media = null;
+  const settings = {
+    mediaTypes: MediaGroups.createMediaTypes(),
+    masterPlaylistLoader: {
+      media: () => media
+    }
+  };
+  const groups = settings.mediaTypes[type].groups;
+  const tracks = settings.mediaTypes[type].tracks;
+  const activeTrack = MediaGroups.activeTrack[type](type, settings);
+  const activeGroup = MediaGroups.activeGroup(type, settings);
+
+  assert.equal(activeGroup(), null, 'returns null when no media in masterPlaylistLoader');
+
+  media = { attributes: { } };
+
+  // there is no default `main` group for subtitles like there is for audio
+  assert.notOk(activeGroup(), 'returns null when media does not specify subtitle group');
```

Comment:
not sure what you mean

Reply:
I was thinking that there could be a subtitle group, but none of the renditions actually reference it, but after re-reading the test I don't think it would make sense contextually here, and I doubt we'll run into an issue with it being in a manifest.